sidecar/internal/service/reclaimspace.go

```diff
@@ -61,12 +66,29 @@ func (rs *ReclaimSpaceServer) ControllerReclaimSpace(
 	pvName := req.GetPvName()
 	klog.Info(pvName)
 
-	// TODO: get following params using kubeClient and pvName
+	var deviceSecret map[string]string
+	var volID string
+	var volAttributes map[string]string
+
+	pv, err := rs.kubeClient.CoreV1().PersistentVolumes().Get(context.TODO(), pvName, metav1.GetOptions{})
+	if pv.Spec.CSI != nil {
+		volID = pv.Spec.CSI.VolumeHandle
+		volAttributes = pv.Spec.CSI.VolumeAttributes
+
+		if pv.Spec.CSI.NodeStageSecretRef != nil {
+			// Get the secrets from the k8s cluster
+			deviceSecret, err = kube.GetSecret(ctx, rs.kubeClient, pv.Spec.CSI.NodeStageSecretRef.Name, pv.Spec.CSI.NodeStageSecretRef.Namespace)
+			if err != nil {
+				klog.Errorf("Failed to get secret %v", err)
+				return nil, status.Error(codes.InvalidArgument, err.Error())
+			}
+		}
+	}
 
 	csiReq := &csiReclaimSpace.ControllerReclaimSpaceRequest{
-		VolumeId:   "",
-		Parameters: map[string]string{},
-		Secrets:    map[string]string{},
+		VolumeId:   volID,
+		Parameters: volAttributes,
+		Secrets:    deviceSecret,
```

Comment:
you can create `csiReq` earlier in the function, and only set `VolumeID` there. Then
```
if pv.Spec.CSI.NodeStageSecretRef != nil {
    csiReq.Secrets, err = kube.GetSecret(...)
}

res, err := rs.controllerClient.ControllerReclaimSpace(ctx, csiReq)
```

this hopefully makes the function a little simpler.